### ISSUE TITLE
Mandatory OAuth2.0

### ISSUE THREAD

**Issue by RichieRoxx** [AMSoftwareNL/crmpowershell#27]
Hey,

have you ever ecountered Environments with mandatory OAuth2.0-Authentication? I just did... :-/

https://docs.microsoft.com/en-us/powerapps/developer/data-platform/authenticate-dot-net-framework

Steps to reproduce
------------------
Connect-CrmDeployment -Credential $credential -Region $CRMRegion -Verbose

Expected behavior
-----------------
Connection to CrmDEployment ;) 

Actual behavior
---------------
```
Connect-CrmOrganization : An error occurred when processing the security tokens in the message:You are using Ws-Tust authentication which has been deprecated and no longer 
supported in your environment. Please use oAuth2.0 authentication.
In C:\myscript.ps1:22 Zeichen:1
+ Connect-CrmOrganization $OrgUniqueName
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Connect-CrmOrganization], FaultException
    + FullyQualifiedErrorId : AMSoftware.Crm.PowerShell.Commands.Discovery.ConnectOrganizationCommand

```
Environment data
----------------

**Comment by AMSoftwareNL** [comment]
I was afraid this was going to happen sooner or later, and therefore already created issue #24. 
Will try to fix this as soon as possible, with maybe some enhancements. One I'm thinking about is to be able to use Accesstoken and/or Service Endpoint from Azure DevOps.

**Comment by RichieRoxx** [comment]
I'll see that I get some more info on that. The environment I try to connect to is also secured through Azure conditional access. I could connect with the xrmToolBox by using the Oauth/MFA-Authentication Flow, if there is something to learn from that?!
![image](https://user-images.githubusercontent.com/17089971/111631208-6cf91d00-87f3-11eb-8e42-c300473c2f5c.png)

**Comment by RichieRoxx** [comment]
I have a bit more info on this topic now. So the deprecation of ws-trust has been announced for a while:
https://docs.microsoft.com/en-us/power-platform/important-changes-coming#deprecation-of-office365-authentication-type-and-organizationserviceproxy-class-for-connecting-to-dataverse

The actual enforcement comes into play now more an more ("authentication protocol" means ws-trust):
To allow for transition of customers and partner applications:

1. Effective October 2020, the authentication protocol will be retired for all new tenants.
2. Effective October 2020, the authentication protocol will not be available in all new regions.
3. Effective April 2021*, the authentication protocol will be retired for all new environments within a tenant.
4. Effective April 2022, the authentication protocol will be retired for all new and existing environments within a tenant.

*The retirement of new environments starts April 1st, 2021 and rolls out gradually in all regions within a 6-week window.

Case 1. got me with a customer just recently and case 3. is about to strike very soon. We have implemented OAuth in our own software recently and I might be able to make PR based on that in April.

**Comment by RichieRoxx** [comment]
Also with the api-limits on the horizon, there should be even more motivation to move to an AppReg/Application User-Approach, since an Application user is granted 100.000 Requests per 24hrs, compared to 20.000 for a normal user:

https://docs.microsoft.com/en-us/power-platform/admin/api-request-limits-allocations

https://docs.microsoft.com/en-us/powerapps/developer/common-data-service/api-limits

**Comment by AMSoftwareNL** [comment]
Resolved in v2.0.0.0
